packages/htmlviewer-extension/schema/plugin.json

```diff
@@ -0,0 +1,71 @@
+{
+  "title": "HTML Viewer",
+  "description": "HTML Viewer settings.",
+  "jupyter.lab.toolbars": {
+    "HTML Viewer": [
+      { "name": "refresh", "rank": 10 },
+      { "name": "trust", "rank": 20 }
+    ]
+  },
+  "jupyter.lab.transform": true,
+  "properties": {
+    "toolbar": {
+      "title": "HTML viewer toolbar items",
+      "description": "Note: To disable a toolbar item,\ncopy it to User Preferences and add the\n\"disabled\" key. The following example will disable the refresh item:\n{\n  \"toolbar\": [\n    {\n      \"name\": \"refresh\",\n      \"disabled\": true\n    }\n  ]\n}\n\nToolbar description:",
+      "items": {
+        "$ref": "#/definitions/toolbarItem"
+      },
+      "type": "array",
+      "default": []
+    }
+  },
+  "additionalProperties": false,
+  "type": "object",
+  "definitions": {
+    "toolbarItem": {
+      "properties": {
+        "name": {
+          "title": "Unique name",
+          "type": "string"
+        },
+        "args": {
+          "title": "Command arguments",
+          "type": "object"
+        },
+        "command": {
+          "title": "Command id",
+          "type": "string",
+          "default": ""
```

Comment:
I wonder if we could publish our own types under some URL to avoid repeating this. It might make updating to new versions easier as well.

Reply:
Indeed this would be nice. The true source should probably the settings schema. Do you know how and where to publish such schema ?

Reply:
One option would be https://www.schemastore.org/json/ (I think they would accept one or two shema files for JupyterLab but we should not be publishing too many files there). Another potential option would be to publish it on jupyter.org which might be more reliable long-term but would need a few people to agree (and more work). For proof-of-concept I would start by linking to one of the settings schema on GitHub (via raw.githubusercontent.com or similar to get the raw file) and see if the references work.

Reply:
I will open a follow-up issue on that point. One trouble that may arise from that is the support of offline installation. But let's discuss that further in that follow-up.